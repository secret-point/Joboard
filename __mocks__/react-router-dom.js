// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
