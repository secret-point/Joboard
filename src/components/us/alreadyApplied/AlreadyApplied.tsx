import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { redirectToDashboard } from "../../../helpers/utils";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { translate as t } from "../../../utils/translator";

interface MapStateToProps {
  candidate: CandidateState
}

const AlreadyApplied = (props: MapStateToProps) => {
  const { candidate } = props;
  const { candidateData } = candidate.results;
  const firstName = candidateData?.firstName;

  useEffect(() => {
    boundGetCandidateInfo();
  }, [])

  const handleGoToDashboard = () => {
    redirectToDashboard();
  }

  return (
    <Col gridGap="S300" padding={{ top: 'S300' }}>
      <Text>
        {firstName ? `${firstName}, `: ''}
        {t("BB-already-applied", "You have already started and / or completed an application for this job.")}
      </Text>
      <Text fontSize="T100">
        {t("BB-already-applied-description", "Return to the dashboard resume your application or view application and new hire appointment details, and your next steps.")}
      </Text>
      <Button variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
        {t("BB-already-applied-button-text", "Return to dashboard")}
      </Button>
    </Col >
  )
}

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AlreadyApplied);
