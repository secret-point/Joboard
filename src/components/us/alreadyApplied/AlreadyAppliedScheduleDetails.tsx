import React from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import moment from "moment";
import { PayRateType } from "../../../countryExpansionConfig";
import { localeToLanguageList } from "../../../utils/constants/common";
import { FEATURE_FLAG } from "../../../utils/enums/common";
import {
  formatMonthlyBasePayHelper,
  getCountryCode,
  getFeatureFlagValue,
  getLocale,
  getPayRateCountryConfig,
  getSpanishLocaleDateFormatter,
  showRequiredLanguageByCountry
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { Schedule } from "../../../utils/types/common";
import { Card } from "@amzn/stencil-react-components/card";

interface ScheduleDetailsProps {
  scheduleDetail: Schedule;
}

export const AlreadyAppliedScheduleDetails = (props: ScheduleDetailsProps) => {

  const { scheduleDetail } = props;
  const requiredLanguages = scheduleDetail?.requiredLanguage || [];
  const showRequiredLanguages = getFeatureFlagValue(FEATURE_FLAG.MLS) && requiredLanguages?.length > 0;

  const {
    hoursPerWeek,
    firstDayOnSite,
    currencyCode,
    scheduleText,
    totalPayRate,
    totalPayRateL10N,
    monthlyBasePay,
    monthlyBasePayL10N,
    scheduleBannerText
  } = scheduleDetail;

  const renderStartDate = () => {
    const startDate = `${moment(firstDayOnSite).locale(getLocale()).format("MMM DD, YYYY")}`;
    return getSpanishLocaleDateFormatter(startDate);
  };

  const renderRequiredLanguages = (): string => {
    const languageList: string[] = [];

    localeToLanguageList.forEach((item) => {
      requiredLanguages?.includes(item.locale) ? languageList.push(t(item.translationKey, item.language)) : null;
    });

    return languageList.join(", ");
  };

  const renderPayRateByLocale = () => {
    if (getPayRateCountryConfig(getCountryCode()).payRateType === PayRateType.monthMax) {
      const monthlyRate = monthlyBasePayL10N ? monthlyBasePayL10N : formatMonthlyBasePayHelper(monthlyBasePay, currencyCode);
      const formattedMonthlyRate = currencyCode && monthlyRate ? `${currencyCode}${monthlyRate}` : null;
      return formattedMonthlyRate ? t("BB-Schedule-card-total-pay-per-month-text", `Up to ${formattedMonthlyRate}/month`, { formattedMonthlyRate }) : t("BB-Schedule-card-not-applicable", "N/A");
    }
    return `${totalPayRateL10N || currencyCode + totalPayRate}`;

  };

  return (
    <Card padding="S300" flexDirection="column">
      <Col gridGap="S200">
        <Row gridGap="S300" alignItems="center">
          <Text fontSize="T100" fontWeight="bold">{t("BB-Schedule-card-hours", "Hours")}: </Text>
          <Text fontSize="T100">{hoursPerWeek} {t("BB-Schedule-card-hours-per-week-text", "hours/week")}</Text>
        </Row>

        <Row gridGap="S300" alignItems="center">
          <Text fontSize="T100">{scheduleText}</Text>
        </Row>

        {showRequiredLanguages && showRequiredLanguageByCountry(getCountryCode()) && (
          <Row gridGap="S300" alignItems="center">
            <Text fontSize="T100" fontWeight="bold">
              {t("BB-Schedule-card-languages-supported", "Languages Supported")}:
            </Text>
            <Text fontSize="T100">{renderRequiredLanguages()}</Text>
          </Row>
        )}
        <Row gridGap="S300" alignItems="center">
          <Row width="65%" gridGap="S300">
            <Text fontSize="T100" fontWeight="bold">{t("BB-Schedule-card-start-date", "Start Date")}: </Text>
            <Text fontSize="T100">{renderStartDate()}</Text>
          </Row>
          <Row width="%35">
            <Text fontSize="T300" fontWeight="bold">{renderPayRateByLocale()}</Text>
            <Row alignSelf="end"><Text fontSize="T100">{t("BB-Schedule-card-total-pay-per-hour-text", "/hour")}</Text></Row>
          </Row>
        </Row>
        { scheduleBannerText && (
          <Row padding={{ top: "S300" }} className="scheduleBannerText">
            <Text fontSize="T100">{scheduleBannerText}</Text>
          </Row>
        )}
      </Col>
    </Card>
  );
};

export default AlreadyAppliedScheduleDetails;
