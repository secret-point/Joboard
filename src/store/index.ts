import { createStore, applyMiddleware, compose  } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";
import { createHashHistory } from "history";
import { routerMiddleware  } from "react-router-redux";
import { createLogger } from "redux-logger";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const middleware = [thunk, routerMiddleware(createHashHistory()), createLogger()];
const store = createStore(rootReducer, {} , composeEnhancers(applyMiddleware(...middleware)));
export default store;