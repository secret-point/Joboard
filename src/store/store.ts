import { applyMiddleware, createStore } from "redux";
import { routerMiddleware } from "connected-react-router";
import { createHashHistory } from "history";
import { createEpicMiddleware } from "redux-observable";
import createRootReducer from "../reducers/root.reducer";
import rootEpic from "../epics/root.epic";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage'

/* TODO remove any from functions */
const logger = () => (next: any) => (action: any) => {
  // eslint-disable-next-line no-console
  // console.log("dispatching", action);
  const result = next(action);
  // console.log("next state", store.getState());
  // eslint-disable-next-line no-console
  return result;
};

export const history = createHashHistory();

const persistConfig = {
  key: 'root',
  storage, // defaults to localStorage for web
  whitelist: ['request']
};

export const configureStore = (preloadedState: any) => {
  const epicMiddleware = createEpicMiddleware();
  const middlewares: any = [epicMiddleware, routerMiddleware(history), logger];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer];
  const composedEnhancers: any = composeWithDevTools(...enhancers);
  const persistedReducer = persistReducer(persistConfig, createRootReducer(history));

  const store = createStore(
    persistedReducer,
    preloadedState,
    composedEnhancers
  );

  epicMiddleware.run(rootEpic);

  return store;
};

const store = configureStore({});

export const persistor = persistStore(store);

export default store;
