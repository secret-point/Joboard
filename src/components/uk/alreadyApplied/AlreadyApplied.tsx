import React, { useEffect } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { getPageNameFromPath, parseQueryParamsArrayToSingleItem, redirectToDashboard } from "../../../helpers/utils";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { getLocale } from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { DUPLICATE_JOB_ID_LOCAL_STORAGE_KEY } from "../../../utils/constants/common";

interface MapStateToProps {
  candidate: CandidateState;
  job: JobState;
}

export const AlreadyApplied = (props: MapStateToProps) => {
  const { candidate, job } = props;
  const { candidateData } = candidate.results;
  const firstName = candidateData?.firstName;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const pageName = getPageNameFromPath(pathname);
  const { jobId } = queryParams;
  const jobDetail = job.results;

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && addMetricForPageLoad(pageName);
  }, [jobDetail, pageName]);

  const handleGoToDashboard = () => {
    localStorage.setItem(DUPLICATE_JOB_ID_LOCAL_STORAGE_KEY, jobId);
    redirectToDashboard();
  };

  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>
      <Text>
        {firstName ? `${firstName}, `: ""}
        {t("BB-already-applied-UK", "Thank you for applying to Amazon!")}
      </Text>
      <Text fontSize="T100">
        {t("BB-already-applied-description-UK", "You have already applied to this job. Return to the dashboard to view your application.")}
      </Text>
      <Col padding={{ top: "S300" }}>
        <Button dataTestId="button-dashboard" variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
          {t("BB-already-applied-button-text-UK", "Go to dashboard")}
        </Button>
      </Col>
    </Col>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AlreadyApplied);
