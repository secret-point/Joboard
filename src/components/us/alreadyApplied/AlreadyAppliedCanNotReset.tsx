import React, { useEffect } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col, Flex, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { getPageNameFromPath, parseQueryParamsArrayToSingleItem, redirectToDashboard } from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { getLocale } from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { DuplicateJobId } from "../../../utils/constants/common";
import { useBreakpoints } from "@amzn/stencil-react-components/responsive";

interface MapStateToProps {
  job: JobState;
}

export const AlreadyAppliedCanNotReset = (props: MapStateToProps) => {
  const { job } = props;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const pageName = getPageNameFromPath(pathname);
  const { jobId } = queryParams;
  const jobDetail = job.results;
  const { matches } = useBreakpoints();

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && addMetricForPageLoad(pageName);
  }, [jobDetail, pageName]);

  const handleGoToDashboard = () => {
    sessionStorage.setItem(DuplicateJobId, jobId);
    redirectToDashboard();
  };

  return (
    <Col gridGap="S300" padding={matches.l || matches.xl ? { top: "S300", left: "450px" } : {}}>
      <Text fontSize="T400" fontWeight="bold">
        {t("BB-already-applied-v2", `You have already submitted an application for the role:  ${jobDetail?.jobTitle || ""}`, { jobTitle: jobDetail?.jobTitle || "" })}
        <br />
        {jobDetail?.jobTitle || ""}
      </Text>
      <Text fontSize="T300" color="neutral70">
        {t("BB-job-id", "Job Id: ")} {jobDetail?.jobId || "" }
      </Text>
      <Row width={matches.s || matches.m ? "100%": "60%"}>
        <Text fontSize="T200">
          {t("BB-already-applied-description-v2", "You can only have one active application for this Job Id. To view your current application and see more options, click 'Go to dashboard'.")}
        </Text>
      </Row>
      <Flex className="alreadyAppliedButton" flexDirection={matches.s || matches.m ? "column": "row"}>
        <Button dataTestId="button-go-to-dashboard" variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
          {t("BB-already-applied-button-text-v2", "Go to dashboard")}
        </Button>
      </Flex>
    </Col>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AlreadyAppliedCanNotReset);
