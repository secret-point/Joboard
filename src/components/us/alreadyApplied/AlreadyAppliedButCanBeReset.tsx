import React, { useEffect } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col, Flex, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { getPageNameFromPath, parseQueryParamsArrayToSingleItem, redirectToDashboard } from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { useBreakpoints } from "@amzn/stencil-react-components/responsive";
import {
  checkAndBoundGetApplication, createUpdateApplicationRequest,
  getLocale,
  routeToAppPageWithPath
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { AlreadyAppliedScheduleDetails } from "./AlreadyAppliedScheduleDetails";
import { Application } from "../../../utils/types/common";
import { onCompleteTaskHelper } from "../../../actions/WorkflowActions/workflowActions";
import { QUERY_PARAMETER_NAME, UPDATE_APPLICATION_API_TYPE, WORKFLOW_STEP_NAME } from "../../../utils/enums/common";
import { ApplicationState } from "../../../reducers/application.reducer";
import { PAGE_ROUTES } from "../../pageRoutes";
import { boundHidePageLoader } from "../../../actions/UiActions/boundUi";
import {
  boundUpdateApplicationDS
} from "../../../actions/ApplicationActions/boundApplicationActions";
import { SelectedScheduleForUpdateApplication } from "../../../utils/apiTypes";
import { DuplicateJobId } from "../../../utils/constants/common";
import { log, LoggerType } from "../../../helpers/log-helper";
import { AppConfigState } from "../../../reducers/appConfig.reducer";

interface MapStateToProps {
  job: JobState;
  schedule: ScheduleState;
  application: ApplicationState;
  appConfig: AppConfigState;
}

export const AlreadyAppliedButCanBeReset = (props: MapStateToProps) => {
  const { application, job, schedule, appConfig } = props;
  const { search, pathname } = useLocation();
  const { matches } = useBreakpoints();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const pageName = getPageNameFromPath(pathname);
  const { scheduleId, jobId } = queryParams;
  const jobDetail = job.results;
  const { scheduleDetail, oldScheduleDetail } = schedule.results;
  const { errorMetadata } = application;

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    scheduleId && scheduleId !== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleId]);

  useEffect(() => {
    errorMetadata?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: errorMetadata?.scheduleId,
      isOldSchedule: true
    });
    boundHidePageLoader();
  }, [errorMetadata]);

  useEffect(() => {
    jobDetail && addMetricForPageLoad(pageName);
  }, [jobDetail, pageName]);

  const handleGoToDashboard = () => {
    sessionStorage.setItem(DuplicateJobId, jobId);
    redirectToDashboard(jobDetail?.jobId);
  };

  const handleContinueToApplication = () => {
    if (errorMetadata?.applicationId && errorMetadata?.candidateId) {
      const application = {
        applicationId: errorMetadata.applicationId,
        candidateId: errorMetadata.candidateId,
        jobScheduleSelected: { jobId: jobId }
      } as Application;
      window.hasCompleteTaskOnWorkflowConnect = () => {
        onCompleteTaskHelper(application, true, WORKFLOW_STEP_NAME.RESET_APPLICATION, undefined);
      };
      if (scheduleId && appConfig.results?.envConfig) {
        window.hasCompleteTaskOnSkipSchedule = () => {
          const selectedSchedule: SelectedScheduleForUpdateApplication = {
            jobId,
            scheduleId,
            scheduleDetails: JSON.stringify(scheduleDetail)
          };
          const request = createUpdateApplicationRequest(application, UPDATE_APPLICATION_API_TYPE.JOB_CONFIRM, selectedSchedule);
          boundUpdateApplicationDS(request, (applicationResponse: Application) => {
            onCompleteTaskHelper(applicationResponse, undefined, undefined, WORKFLOW_STEP_NAME.JOB_OPPORTUNITIES);
          });
        };
      }
      routeToAppPageWithPath(PAGE_ROUTES.ALREADY_APPLIED, [
        { paramName: QUERY_PARAMETER_NAME.APPLICATION_ID, paramValue: errorMetadata.applicationId }]);

      checkAndBoundGetApplication(errorMetadata.applicationId);
    } else {
      log("errorMetadata,candidateId or applicationId is null/undefined", errorMetadata, LoggerType.INFO);
    }
  };

  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>
      <Text fontSize="T300" fontWeight="bold">
        {t("BB-already-applied-but-can-be-reset", "You already have an active application for the role:",)}
        <br />
        {jobDetail?.jobTitle || ""}
      </Text>
      <Flex flexDirection={matches.s || matches.m ? "column": "row"} gridGap="S300">
        <Col width={matches.s || matches.m ? "100%": "35%"} gridGap="S300">
          <div>{t("BB-existing-application", "Existing application:")}</div>
          {oldScheduleDetail && <AlreadyAppliedScheduleDetails scheduleDetail={ oldScheduleDetail } />}
        </Col>

        <Col width={matches.s || matches.m ? "100%": "35%"} gridGap="S300">
          <div>{t("BB-new-application", "New application:")}</div>
          {scheduleDetail && <AlreadyAppliedScheduleDetails scheduleDetail={ scheduleDetail } />}
        </Col>
      </Flex>
      <Row>
        <Text fontSize="T100">
          {t("BB-already-applied-but-can-be-reset-dashboard-description", "If you proceed with this new application, your existing one will be withdrawn. To view or resume your existing application, click 'Go to dashboard'.")}
        </Text>
      </Row>
      <Row>
        <Text fontSize="T100">
          {t(" BB-already-applied-but-can-be-reset-continue-description", "To proceed with new application, click 'Continue with application'.")}
        </Text>
      </Row>
      <Flex className="alreadyAppliedButtonStickyAtBottom" width={matches.s || matches.m ? "100%": "70%"} gridGap={20} justifyContent="flex-end" flexDirection={matches.s || matches.m ? "column": "row"}>
        <Button dataTestId="button-dashboard" variant={ButtonVariant.Secondary} onClick={handleGoToDashboard}>
          {t("BB-already-applied-button-text", "Go to dashboard")}
        </Button>
        <Button dataTestId="button-continue" variant={ButtonVariant.Primary} onClick={handleContinueToApplication}>
          {t("BB-already-applied-but-continue-button-text", "Continue with new application")}
        </Button>
      </Flex>
    </Col>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(AlreadyAppliedButCanBeReset);
