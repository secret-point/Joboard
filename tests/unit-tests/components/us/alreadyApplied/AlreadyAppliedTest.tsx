import React from "react";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";
import { ReactWrapper, shallow } from "enzyme";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { Router, useLocation } from "react-router-dom";
import { AlreadyApplied } from "../../../../../src/components/us/alreadyApplied/AlreadyApplied";
import store from "../../../../../src/store/store";
import { TEST_APP_CONFIG, TEST_CANDIDATE_STATE, TEST_JOB2_ID, TEST_JOB_STATE } from "../../../../test-utils/test-data";

describe("AlreadyApplied", () => {
  const initStoreState = store.getState();
  let wrapper: ReactWrapper;
  let state: any;
  store.getState = () => state;

  const mockLocation = {
    pathname: "/already-applied",
    search: `?jobId=${TEST_JOB2_ID}`,
    hash: '',
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  beforeEach(() => {
    window.location.assign = jest.fn();

    state = {
      ...initStoreState,
      appConfig: {
        ...initStoreState.appConfig,
        results: {
          ...initStoreState.appConfig.results,
          envConfig: TEST_APP_CONFIG
        }
      }
    };

    const history = createMemoryHistory();
    history.location = {
      ...history.location,
      ...mockLocation
    };

    wrapper = mountWithStencil(
      <Provider store={store}>
        <Router history={history}>
          <AlreadyApplied candidate={TEST_CANDIDATE_STATE} job={TEST_JOB_STATE} />
        </Router>
      </Provider>
    );
  });

  it("should match snapshot", () => {
    const shallowWrapper = shallow(<AlreadyApplied candidate={TEST_CANDIDATE_STATE} job={TEST_JOB_STATE} />);

    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should redirect to dashboard if click on the return to dashboard button", () => {
    const button = wrapper.find('button[data-test-id="button-dashboard"]');
    button.simulate('click');

    expect(window.location.assign).toBeCalledWith(
      expect.stringContaining("/myApplications")
    );
  });
});
