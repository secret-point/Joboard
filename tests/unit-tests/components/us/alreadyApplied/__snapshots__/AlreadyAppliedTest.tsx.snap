// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlreadyApplied should match snapshot for APPLICATION_ALREADY_EXIST 1`] = `
<Col>
  <Connect(AlreadyAppliedCanNotReset) />
</Col>
`;

exports[`AlreadyApplied should match snapshot for APPLICATION_ALREADY_EXIST_CAN_BE_RESET 1`] = `
<Col>
  <Connect(AlreadyAppliedButCanBeReset) />
</Col>
`;
